### ISSUE TITLE
Build failed on linux-4.11

### ISSUE THREAD

**Issue by gyted** [cormander/tpe-lkm#20]
Hi! First of all thank you for maintaining this.

I tried build it on Archlinux with kernel 4.11 but it failed. Here log output:
```
./scripts/find_kernel_src.sh: line 8: arch: command not found
make -C /lib/modules/4.11.0-1/build M=/tmp/build/tpe-lkm/src/tpe-lkm modules
make[1]: Entering directory '/usr/lib/modules/4.11.0-1/build'
  CC [M]  /tmp/build/tpe-lkm/src/tpe-lkm/fopskit.o
  CC [M]  /tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.o
  CC [M]  /tmp/build/tpe-lkm/src/tpe-lkm/tpe_module.o
  CC [M]  /tmp/build/tpe-lkm/src/tpe-lkm/tpe_config.o
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c: In function ‘tpe_file_getfattr’:
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c:18:18: error: ‘const struct inode_operations’ has no member named ‘getxattr’; did you mean ‘getattr’?
  if (!inode->i_op->getxattr) return 0;
                  ^~
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c:20:19: error: ‘const struct inode_operations’ has no member named ‘getxattr’; did you mean ‘getattr’?
  ret = inode->i_op->getxattr(get_dentry(file),
                   ^~
In file included from /tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c:2:0:
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c: In function ‘tpe_log_denied_action’:
/tmp/build/tpe-lkm/src/tpe-lkm/tpe.h:22:49: error: dereferencing pointer to incomplete type ‘const struct cred’
 #define get_task_uid(task) __kuid_val(task->cred->uid)
                                                 ^
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c:83:3: note: in expansion of macro ‘get_task_uid’
   get_task_uid(current),
   ^~~~~~~~~~~~
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c: In function ‘tpe_allow_file’:
/tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.c:148:21: error: implicit declaration of function ‘in_group_p’ [-Werror=implicit-function-declaration]
  if (tpe_dmz_gid && in_group_p(KGIDT_INIT(tpe_dmz_gid)))
                     ^~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:295: /tmp/build/tpe-lkm/src/tpe-lkm/tpe_core.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:1492: _module_/tmp/build/tpe-lkm/src/tpe-lkm] Error 2
make[1]: Leaving directory '/usr/lib/modules/4.11.0-1/build'
make: *** [Makefile:24: tpe.ko] Error 2
[1m[31m==> ERROR:(B[m[1m A failure occurred in build().(B[m
[1m    Aborting...(B[m
```

I don't know if this is something specific or general error. I would be grateful if you can look at this.

**Comment by pjperry** [comment]
Seems as of kernel-4.8, getxattr inode operations were replaced by helper functions which should be used instead. See here:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/fs/xattr.c?h=v4.11&id=6c6ef9f26e598fb977f60935e109cd5b266c941a

**Comment by cormander** [comment]
The newest kernel I have to test with is 4.10 (Fedora 25), and it appears there is other changes between then and 4.11. Can you apply this patch to remove the "soften" support, and let me know whether it compiles, or throws further errors? I'll be able to have a more in-depth look at a later time.

    diff --git a/tpe_core.c b/tpe_core.c
    index c867666..c911716 100644
    --- a/tpe_core.c
    +++ b/tpe_core.c
    @@ -7,6 +7,7 @@ unsigned long tpe_alert_fyet = 0;
     /* check if there's a security.tpe extended file attribute */

     int tpe_file_getfattr(const struct file *file, const char *method) {
    +#ifdef TPE_XATTR
            char context[MAX_FILE_LEN], buffer[MAX_FILE_LEN], *b, *c;
            char attr[MAX_FILE_LEN] = "soften_";
            struct inode *inode = get_inode(file);
    @@ -36,6 +37,7 @@ int tpe_file_getfattr(const struct file *file, const char *method) {
                    if (!strncmp(c, attr, (int)strlen(c)))
                            return 1;
            }
    +#endif
    
            return 0;
     }

**Comment by cormander** [comment]
This issue should be resolved by 7b3085202c213bbb10365cf4a00212df8107ab05. Please reopen if that's not the case. Thanks,
